core/adapters/eth_tx_abi_encode_internals_test.go

```diff
@@ -0,0 +1,358 @@
+	},
+	{
+		"valid empty string",
+		`[{"inputs":[{"name":"a","type":"string"}],"name":"foo","type":"function"}]`,
+		`{"a": ""}`,
+		`f31a696900000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000`,
+	},
+	// Uint
+	{
+		"value too large for uint",
+		`[{"inputs":[{"name":"a","type":"uint8"}],"name":"foo","type":"function"}]`,
+		`{"a": "0x100"}`,
+		``,
+	},
+	{
+		"hex without 0x prefix for uint",
+		`[{"inputs":[{"name":"a","type":"uint8"}],"name":"foo","type":"function"}]`,
+		`{"a": "ff"}`,
+		``,
+	},
+	{
+		"value too small for uint",
+		`[{"inputs":[{"name":"a","type":"uint8"}],"name":"foo","type":"function"}]`,
+		`{"a": "-1"}`,
+		``,
+	},
+	{
+		"different encodings for uint",
+		`[{"inputs":[{"name":"a","type":"uint8"}, {"name":"b","type":"uint8"}],"name":"foo","type":"function"}]`,
+		`{"a": "0x00", "b": "255"}`,
+		`7ce60e3b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000ff`,
+	},
+	{
+		"different encodings for uint",
+		`[{"inputs":[{"name":"a","type":"uint8"}, {"name":"b","type":"uint8"}],"name":"foo","type":"function"}]`,
+		`{"a": "0xa", "b": "0"}`,
+		`7ce60e3b000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000`,
+	},
+	{
+		"different encodings of max uint256",
+		`[{"inputs":[{"name":"a","type":"uint256"}, {"name":"b","type":"uint256"}],"name":"foo","type":"function"}]`,
+		`{
+			"a": "115792089237316195423570985008687907853269984665640564039457584007913129639935",
+			"b": "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+		}`,
+		`04bc52f8ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff`,
+	},
+	{
+		"uint48 as number and string",
+		`[{"inputs":[{"name":"a","type":"uint48"},{"name":"b","type":"uint48"}],"name":"foo","type":"function"}]`,
+		`{"a": 140737488355328, "b": "0x800000000001"}`,
+		`183cb15b00000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000800000000001`,
+	},
+	{
+		"uint56 doesn't accept numbers",
+		`[{"inputs":[{"name":"a","type":"uint56"}],"name":"foo","type":"function"}]`,
+		`{"a": 1"}`,
+		``,
+	},
+}
+
+func TestEthTxABIEncodeAdapter_encode(t *testing.T) {
+	for i, test := range encodeTests {
+		testABI, err := abi.JSON(strings.NewReader(test.abiJSON))
+		assert.NoError(t, err)
+
+		// there should be a single method, get its abi
+		assert.Equal(t, 1, len(testABI.Methods))
+		var fnABI abi.Method
+		for _, fnABI = range testABI.Methods {
+		}
+
+		args, ok := gjson.Parse(test.resultJSON).Value().(map[string]interface{})
+		assert.True(t, ok, "Failed to parse resultJSON in test %v: %s", i, test.desc)
+
+		encoded, err := abiEncode(&fnABI, args)
+		_ = encoded
+		if test.hexEncoded != "" {
```

Comment:
We've been trying to move away from `if` statements in tests. Could you break out success and error conditions into two different sets of tests?

Reply:
Done